### PR TITLE
Fork field-caps for ES|QL

### PR DESCRIPTION
**Pull request by dnhatn**
We need to fork the field-caps API for ES|QL to allow changes to the new internal API without risking breaking the external field-caps API.

### CODE REVIEW

**Review by nik9000 — COMMENTED** [review]
I think this is a good direction to go in.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/IndexPrivilege.java [hunk 1]
```diff
@@ -86,6 +86,7 @@ public final class IndexPrivilege extends Privilege {
         TransportClusterSearchShardsAction.TYPE.name(),
         TransportSearchShardsAction.TYPE.name(),
         TransportResolveClusterAction.NAME,
+        "indices:data/read/esql/resolve_fields",
```
Comment (nik9000) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/IndexPrivilege.java:89]: @jakelandis should double check this one.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlResolveFieldsAction.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import org.elasticsearch.action.fieldcaps.FieldCapabilitiesResponse;
+import org.elasticsearch.action.fieldcaps.TransportFieldCapabilitiesAction;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.HandledTransportAction;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.util.concurrent.EsExecutors;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.transport.TransportService;
+
+public class EsqlResolveFieldsAction extends HandledTransportAction<FieldCapabilitiesRequest, FieldCapabilitiesResponse> {
```
Comment (nik9000) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlResolveFieldsAction.java:22]: Wants javadoc I think. So we know it's our fork.

**Review by nik9000 — APPROVED** [review]

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlResolveFieldsAction.java (hunk 2 above)
Comment (dnhatn) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlResolveFieldsAction.java:27]: ++, added in https://github.com/elastic/elasticsearch/pull/110738/commits/2705ed67ad9420d00cf68970a8b4ceed06693971

**Review by astefan — APPROVED** [review]
LGTM

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlFeatures.java [hunk 3]
```diff
@@ -174,6 +174,11 @@ public class EsqlFeatures implements FeatureSpecification {
      */
     public static final NodeFeature METRICS_SYNTAX = new NodeFeature("esql.metrics_syntax");
 
+    /**
+     * New internal resolve_fields API for ES|QL
+     */
+    public static final NodeFeature NEW_RESOLVE_FIELDS_API = new NodeFeature("esql.new_resolve_fields_api");
```
Comment (jakelandis) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlFeatures.java:180]: nit: "new" in the names of things tends to age poorly, perhaps resolve_fields_for_esql?

**Review by jakelandis — APPROVED** [review]
LGTM (sorry for the delay)

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlFeatures.java (hunk 3 above)
Comment (dnhatn) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlFeatures.java]: ++ I pushed https://github.com/elastic/elasticsearch/pull/110738/commits/4ec1f8005510c9234a9986def3d635e34003321f

**Comment by dnhatn** [comment]
Thanks friends!

**Pull request merged**